src/main/service/GeocodingService.ts

```diff
@@ -0,0 +1,40 @@
+import nodeGeocoder, { Entry, Geocoder, Options } from 'node-geocoder';
+import { GeocodingObject } from '../data/GeocodingObject';
+import * as dotenv from 'dotenv';
+
+/**
+ * A Node library for geocoding our restaurant addresses
+ */
+export class GeocodingService {
+  private geocoder: Geocoder;
+
+  constructor() {
+    dotenv.config();
+    const options: Options = {
+      provider: 'locationiq',
+      apiKey: process.env.GEOCODING_KEY,
+    };
+    this.geocoder = nodeGeocoder(options);
+  }
+
+  public getGeoCoordinatesFromAddress(
+    address: string
+  ): Promise<GeocodingObject> {
+    return this.geocoder
+      .geocode(address)
+      .then((entries: Entry[]) => {
+        const result: GeocodingObject = {};
+
+        if (entries?.length) {
+          const geocodeResult = entries[0];
```

Comment:
So first one is always the most accurate?

Reply:
I couldn't find any documentation on whether or not it was the most accurate so I just decided to get the first one

Reply:
hmm let's see if we can spend some time trying to figure out what's the most accurate! There's different ways we can approach this
- one idea we could do is make the request body take in google places lat lon, and compare the lat lon values with those ones and see which ones the closest. But I'll leave it to you to explore and try out which you think is best! I think taking the first one is not the best idea unless its the most accurate

Reply:
Just did a quick search on this and it looks like there are a few options: https://stackoverflow.com/questions/18883601/function-to-calculate-distance-between-two-coordinates

tl;dr either use the Haversine formula (I think we used this in d3 for cpsc 310) or google maps also has a function for it, but that includes installing it in the BE too https://developers.google.com/maps/documentation/javascript/reference#spherical

Reply:
Yeahh for this approach we would need the request body to take in the lat lon from google places, I'm okay with using the first tbh, I think we can refactor and consider a more complex approach once we see an actual bug where the lat/lon is completely off

Reply:
Thanks to the both of you for researching more into it 😄  I can take a deeper look into it this week! If it takes too long based on timeline, since we're trying to merge everything by this weekend, I might just stick with first one but will keep you posted via Slack!